pyspectrafuse/commands/build_cluster_db.py

```
@@ -0,0 +1,390 @@
+        n_clusters = len(best_keys)
+        best_keys['orig_scan'] = best_keys['orig_scan'].astype(int)
+        logger.info(f"Best PSMs: {n_clusters} clusters")
+    finally:
+        conn.close()
+
+    # Merge with stats/purity early (small — just scalars)
+    best_keys = best_keys.merge(stats_purity, on='cluster_id', how='left')
+    best_keys['purity'] = best_keys['purity'].fillna(1.0).astype(np.float32)
+    best_keys['member_count'] = best_keys['member_count'].fillna(1).astype(np.int32)
+    best_keys['project_count'] = best_keys['project_count'].fillna(1).astype(np.int16)
+    del stats_purity; gc.collect()
```

Ruff (0.15.10)

[error] Multiple statements on one line (semicolon)

(E702)